### PR TITLE
feat(connector): add ClientId

### LINKED ISSUE #6972: Connector: add ClientId for independent state management
Add `clientId` support to connector service for state management between multiple clients (Dapp Browser, Chromium extension, Wallet-Connect).

Currently, the connector service uses only the `dAppURL` as the primary key for managing connection state. No way to distinguish between different connection instances.

This task adds `clientId` field to all connector structures and API calls.

Maintain backward compatibility (old clients without `clientId` should continue working)

### PR DESCRIPTION
**Pull request by friofry**
fixes #6972
Allow isolated state management per dapp (Dapp browser, chrome extension, wallet-connect)

Database:
* Added migration to include `client_id` column with composite PRIMARY KEY (url, client_id). Existing records migrated with client_id = '' for backward compatibility.

API:
* Added `ClientID` field to `RPCRequest`, `ConnectorDApp`, `DApp`
* Updated commands 
* `DefaultClientID` constant for backward compatibility. Old clients(browser extension) automatically use empty string "".

### CODE REVIEW

File: services/connector/database/persistence.go [hunk 1]
```diff
@@ -6,29 +6,32 @@ import (
...
 }
 
 func UpsertDApp(db *sql.DB, dApp *DApp) error {
-	_, err := db.Exec(upsertDAppQuery, dApp.URL, dApp.Name, dApp.IconURL, dApp.SharedAccount, dApp.ChainID)
+	_, err := db.Exec(upsertDAppQuery, dApp.URL, dApp.Name, dApp.IconURL, dApp.ClientID, dApp.SharedAccount, dApp.ChainID)
 	return err
 }
 
-func SelectDAppByUrl(db *sql.DB, url string) (*DApp, error) {
+func SelectDAppByUrlAndClientID(db *sql.DB, url string, clientID string) (*DApp, error) {
```
Comment (igor-sirotin) [inline services/connector/database/persistence.go]: `AND` into the function name is a good marker of that something is going wrong 😄 
The arguments are obvious from the arguments list really, let's just drop it:
```suggestion
func SelectDApp(db *sql.DB, url string, clientID string) (*DApp, error) {
```

File: services/connector/api.go [hunk 2]
```diff
@@ -104,12 +104,16 @@ func (api *API) CallRPC(ctx context.Context, inputJSON string) (interface{}, err
 		return nil, fmt.Errorf("method %s is not allowed", request.Method)
 	}
 
-	return api.forwardRPC(ctx, request.URL, request)
+	return api.forwardRPC(ctx, request)
 }
 
 func (api *API) RecallDAppPermission(origin string) error {
```
Comment (igor-sirotin) [inline services/connector/api.go:126]: Let's mark this one as deprecated and make it a pure wrapper of `RecallDAppPermissionV2`

File: services/connector/api.go [hunk 3]
```diff
@@ -104,12 +104,16 @@ func (api *API) CallRPC(ctx context.Context, inputJSON string) (interface{}, err
...
 	// TODO: close the websocket connection
-	return api.c.RecallDAppPermissions(commands.RecallDAppPermissionsArgs{URL: origin})
+	return api.c.RecallDAppPermissions(commands.RecallDAppPermissionsArgs{URL: origin, ClientID: ""})
+}
+
+func (api *API) RecallDAppPermissionWithClientID(origin string, clientID string) error {
```
Comment (igor-sirotin) [inline services/connector/api.go]:
```suggestion
func (api *API) RecallDAppPermissionV2(origin string, clientID string) error {
```

File: services/connector/commands/client_handler.go [hunk 4]
```diff
@@ -126,7 +126,12 @@ func (c *ClientSideHandler) RecallDAppPermissions(args RecallDAppPermissionsArgs
 		return ErrEmptyUrl
 	}
 
-	dApp, err := persistence.SelectDAppByUrl(c.Db, args.URL)
+	// For backward compatibility with old clients(browser extension) that don't provide clientId
+	if args.ClientID == "" {
+		args.ClientID = DefaultClientID
+	}
```
Comment (igor-sirotin) [inline services/connector/commands/client_handler.go]: I think we must not do this, as it introduces a vulnerability.

## Examples

Imagine this case:
1. User permits `OpenSea` dApp with WalletConnect
2. The application is running and exposing the connector service to `localhost`
3. Now any local app can connect to `localhost:8586` and with `ClientID: wallet-connect` and no UI popup will appear.

Transactions would still require manual approval, but at least the list of accounts will be available this way.

## Proposal

`ClientID` should not be set by the client, but only by `status-go` / `status-desktop`. 

So we must set `ClientID` inside "secure environment":
- inside `status-go` - knowing that the connection is coming from HTTP/WS
- inside `status-desktop` (coming from `CallRPC`) - we trust it, because the C-bindings connection is secure enough.

So I think it's closer to `ConnectionType` rather than `ClientID`.

## A problem

There might still be a clash between some `constant httpClientID = "http-client"` that we define in `status-go` and an arbitrary string coming from `status-desktop`.

We can control `status-desktop` client ids, but we can't control all other potential clients of `status-go`. 

So not sure how we should go with this 🤔
Perhaps it's ok if we just define the forbidden constant in status-go docs.

File: services/connector/database/persistence.go (hunk 1 above)
Comment (friofry) [reply services/connector/database/persistence.go]: fixed

File: services/connector/api.go (hunk 2 above)
Comment (friofry) [reply services/connector/api.go:126]: Fixed

File: services/connector/commands/client_handler.go (hunk 4 above)
Comment (friofry) [reply services/connector/commands/client_handler.go]: I agree – passing  `clientId` directly looks like a vulnerability. It would be best to set the `clientId` (connection‑type) at the moment the connection is created

I think `clientId` is still useful because the desktop C‑bindings (will) expose both **wallet‑connect** and **dapp‑browser**.

Proposed approach:

* `http/ws` creates a connection with `connection-type="http/public"`, possibly with a pre‑defined clientId
* For status‑desktop use `connection-type="desktop/secure"`.
* In the `*v2` methods we will allow the caller to pass a `clientId`. The connector will respect the `clientId` only when the connection type is secure.

In other words, use `connection-type` for general protection and `clientId` for state isolation per client.

Additionally, I’d add a session identifier to the `Login` response so that an established connection can’t be hijacked by another application

Comment (friofry) [reply services/connector/commands/client_handler.go]: added a fix

File: services/connector/context.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	if connType, ok := ctx.Value(connectionTypeKey).(ConnectionType); ok {
+		return connType
+	}
+	return ConnectionTypeInternal // default to trusted
+}
+
+// IsUntrustedConnection checks if the connection is from HTTP/WebSocket
+func IsUntrustedConnection(ctx context.Context) bool {
+	return GetConnectionType(ctx) == ConnectionTypeHTTP
+}
```
Comment (igor-sirotin) [inline services/connector/context.go:40]: I think just `IsUntrustedConnection` is enough, why define both 🤔

File: services/connector/context_test.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	"context"
+	"testing"
+)
+
+func TestConnectionType(t *testing.T) {
+	tests := []struct {
+		name          string
+		connType      ConnectionType
+		wantUntrusted bool
+		wantTrusted   bool
```
Comment (igor-sirotin) [inline services/connector/context_test.go]: Aren't these mutually exclusive?

File: services/connector/context_test.go [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+			if got := IsTrustedConnection(ctx); got != tt.wantTrusted {
+				t.Errorf("IsTrustedConnection() = %v, want %v", got, tt.wantTrusted)
+			}
+
+			if got := GetConnectionType(ctx); got != tt.connType {
+				t.Errorf("GetConnectionType() = %v, want %v", got, tt.connType)
+			}
+		})
+	}
```
Comment (igor-sirotin) [inline services/connector/context_test.go]: Mate, keep it simple 😄 
```suggestion
    ctx := WithConnectionType(context.Background(), ConnectionTypeHTTP)
    require.False(IsTrustedConnection(ctx))
    
    ctx = WithConnectionType(context.Background(), ConnectionTypeInternal)
    require.True(IsTrustedConnection(ctx))
```

File: services/connector/api.go [hunk 8]
```diff
@@ -96,6 +97,11 @@ func (api *API) CallRPC(ctx context.Context, inputJSON string) (interface{}, err
 		return "", err
 	}
 
+	// This prevents external clients from spoofing ClientID to impersonate trusted clients
+	if IsUntrustedConnection(ctx) && request.ClientID != "" {
```
Comment (igor-sirotin) [inline services/connector/api.go]:
```suggestion
	if !IsTustedConnection(ctx) && request.ClientID != "" {
```

File: services/connector/commands/client_handler.go (hunk 4 above)
Comment (igor-sirotin) [reply services/connector/commands/client_handler.go]: @friofry looks good, just one thing: in case of `HTTP` connection type you have an empty `ClientID`. So if a _trusted client_ doesn't set any client id, there is still the vulnerability.

Looks like it should be a mix of `type` and `client_id`. 

OR we must forbid empty `client_id` from _trusted clients_.

File: services/connector/context_test.go (hunk 7 above)
Comment (friofry) [reply services/connector/context_test.go]: Indeed:) 

Added more tests for trusted connection + empty clientID check

File: services/connector/context.go (hunk 5 above)
Comment (friofry) [reply services/connector/context.go:40]: Fixed

File: services/connector/commands/client_handler.go (hunk 4 above)
Comment (friofry) [reply services/connector/commands/client_handler.go]: client_id is now mandatory for trusted clients.

**Review by alexjba — APPROVED** [review]

**Review by igor-sirotin — APPROVED** [review]
